backend/src/zimfarm_backend/db/offliner_definition.py

```diff
@@ -0,0 +1,161 @@
+from typing import Any
+from uuid import UUID
+
+from pydantic import BaseModel
+from sqlalchemy import func, select
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm import Session
+
+from zimfarm_backend.common import getnow
+from zimfarm_backend.common.enums import Offliner
+from zimfarm_backend.common.schemas.offliners import create_offliner_schema
+from zimfarm_backend.common.schemas.offliners.builder import OfflinerFlagSchema
+from zimfarm_backend.common.schemas.orms import OfflinerDefinitionSchema
+from zimfarm_backend.db.exceptions import (
+    RecordAlreadyExistsError,
+    RecordDoesNotExistError,
+)
+from zimfarm_backend.db.models import OfflinerDefinition
+
+
+class OfflinerVersionsListResult(BaseModel):
+    """Result of query to list offliner versions from the database."""
+
+    nb_records: int
+    versions: list[str]
+
+
+def create_offliner_definition_schema(
+    offliner_definition: OfflinerDefinition,
+) -> OfflinerDefinitionSchema:
+    """Create the offliner definition schema"""
+    return OfflinerDefinitionSchema(
+        id=offliner_definition.id,
+        offliner=offliner_definition.offliner,
+        version=offliner_definition.version,
+        created_at=offliner_definition.created_at,
+        definition=OfflinerFlagSchema.model_validate(offliner_definition.definition),
+    )
+
+
+def create_offliner_definition(
+    session: Session,
+    definition: OfflinerFlagSchema,
```

Comment:
This `definition: OfflinerFlagSchema` argument makes me realize we probably have something wrong here in terms of naming.

I feel like we should find another name for both the field and the class.

What about switching from `definition` to `schema` (both in DB and in pydantic Schemas), and from `OfflinerFlagSchema` to `OfflinerSchema` (it can and will contain more than only flags at some point)

The `schema` is this dictionary. The `definition` is the whole object.

Or any other name, but reusing `definition` here makes things a bit confusing in the code.